channeldb/channel.go

```diff
@@ -1016,7 +1030,22 @@ func (c *OpenChannel) markBroadcasted(status ChannelStatus, key []byte,
 		}
 	}
 
-	return c.putChanStatus(status, putClosingTx)
+	// If the localCoopInitiated bool is non-nil, generate a closure which
```

Comment:
Imho, it'd be much nicer if you could refactor these tree functions as it's already a bit messy. In this (`markBroadcasted`) function what we really do is create closures which could be helper functions themselves and rather move the locking and the call to `putChanStatus` to `MarkCoopBroadcasted` and `MarkCommitmentBroadcasted`. WDYT?

Reply:
Yeah I think this is a nicer way of going about it, have updated to remove `markBroadcast` entirely in favor of each function creating their own closures & locking.